### PR TITLE
chore(api): Update CRD api versions to v1

### PR DESCRIPTION
**Pull request by fpetkovski**
This PR uses [controller-gen](https://github.com/kubernetes-sigs/controller-tools/tree/master/cmd/controller-gen) in conjunction with [kubebuilder annotations](https://book.kubebuilder.io/reference/markers/crd.html) to update CRD versions to `apiextensions.k8s.io/v1`.

The CRDs can be generated using: `controller-gen crd paths=./apis/metacontroller/... output:stdout > crds.yaml`

### CODE REVIEW

File: hack/get-kube-binaries.sh
Comment (fpetkovski) [inline hack/get-kube-binaries.sh:13]: The api version `apiextensions.k8s.io/v1` is only available from 1.16

File: manifests/metacontroller.yaml
Comment (fpetkovski) [inline manifests/metacontroller.yaml:6]: Turns out the `k8s.io` namespace has been reserved for official Kubernetes resources. Should we maybe apply for approval? 

More information [here](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190612-crd-group-protection.md#what-to-do-if-you-accidentally-put-an-unapproved-api-in-a-protected-group)

**Comment by grzesuav** [comment]
hi @fpetkovski , than for this PR. Currently I have one to upgrade metacontroller dependencies to 0.17 (https://github.com/metacontroller/metacontroller/pull/49) , so after merging that one (I just want to test it locally on minikube, hopefully this happen this week) I would ask for rebasing your (and keep kubernetes dependency for integration test to 1.17.0 to match lib dependencies)

File: manifests/metacontroller.yaml
Comment (grzesuav) [reply manifests/metacontroller.yaml:6]: Thanks, created https://github.com/metacontroller/metacontroller/issues/54

File: hack/get-kube-binaries.sh
Comment (grzesuav) [reply hack/get-kube-binaries.sh:13]: could you change to 1.17.0 as in https://github.com/metacontroller/metacontroller/pull/49

**Comment by grzesuav** [comment]
@fpetkovski As this commit will made metacontroller require k8s at least 1.16, can you add commit body which explicitly states that and starting with `BREAKING CHANGE` message ? 


We use sematic-release for versioning, to mark release as breaking commit body must meet certain conditions (https://github.com/semantic-release/semantic-release#commit-message-format). I would suggest adding :
```
BREAKING CHANGE: Migrated CRD api version to 'apiextensions.k8s.io/v1' introduced in kubernetes 1.16, which is now minimal supported kubernetes version
```

**Comment by fpetkovski** [comment]
Hey @grzesuav thanks for the fast feedback. I've set the Kubernetes version to 1.17 and amended the commit message to indicate that this PR introduces a breaking change.

Let me know as soon as you'd like me to rebase against master.

**Comment by grzesuav** [comment]
@fpetkovski go for rebase, also please check failing check, it basically require you to add `signoff <author>` comment, click on it and it will tell you what to do

**Comment by grzesuav** [comment]
Hmm aparently we have more places where we stick to `v1beta1` - https://github.com/metacontroller/metacontroller/issues/44 , if you have time you can you update also them ? If not, after you rebase I can add another commit a top of yours to have everything in one go.

File: examples/bluegreen/bluegreen-controller.yaml
Comment (grzesuav) [inline examples/bluegreen/bluegreen-controller.yaml:16]: is `served`/`storage` and  `schema` required by CRD version `v1`?

Comment (fpetkovski) [reply examples/bluegreen/bluegreen-controller.yaml:16]: Yes, `schema` is always required. And at least one version must he `served` and `stored`. Since we have only one version in this CRD, the fields should be specified.

**Comment by grzesuav** [comment]
@fpetkovski additional question, I guess it should work with user CRD's in both versions (v1 and v1beta1) but just to be safe, have you tested it ? If not I will test it tommorow

**Comment by fpetkovski** [comment]
@grzesuav yes, v1beta1 and they are essentially treated as different APIs so they can coexist together. However it would be nice if you could also run some basic tests locally so that we don't run into "it works on my machine"

**Comment by grzesuav** [comment]
@fpetkovski can you run locally `make generated_files` on your side and share outcomes ?

File: apis/metacontroller/v1alpha1/types.go
Comment (grzesuav) [inline apis/metacontroller/v1alpha1/types.go:138]: remove this blank like, on `make generated_files` it cause codegen ignores `ControllerRevision` type

**Comment by fpetkovski** [comment]
> @fpetkovski can you run locally `make generated_files` on your side and share outcomes ?

I am getting 
```
+ Generating deepcopy funcs for metacontroller/v1alpha1
F1201 09:24:40.855691   54635 deepcopy.go:131] Failed loading boilerplate: open k8s.io/code-generator/hack/boilerplate.go.txt: no such file or directory
```
But this is also happening on the master branch, not just on this one.

**Comment by grzesuav** [comment]
> > @fpetkovski can you run locally `make generated_files` on your side and share outcomes ?
> 
> I am getting
> 
> ```
> + Generating deepcopy funcs for metacontroller/v1alpha1
> F1201 09:24:40.855691   54635 deepcopy.go:131] Failed loading boilerplate: open k8s.io/code-generator/hack/boilerplate.go.txt: no such file or directory
> ```
> 
> But this is also happening on the master branch, not just on this one.

strange, anyway could you removed the empty line mentioned above ? It makes this step not working correctly.  no idea why it won't work on your side...

**Review by grzesuav — CHANGES_REQUESTED** [review]
- fix the whiteline

I did a bit of testing, using `examples` directory - all seems to work both in case of `v1beta1` and `v1` crd's versions

File: apis/metacontroller/v1alpha1/types.go
Comment (fpetkovski) [reply apis/metacontroller/v1alpha1/types.go:138]: Sure thing, done.

**Review by fpetkovski — COMMENTED** [review]
Thanks @grzesuav, I removed the whiteline. Do you see any additional work that needs to be done?

File: apis/metacontroller/v1alpha1/types.go
Comment (grzesuav) [reply apis/metacontroller/v1alpha1/types.go:138]: I can  still see line 138 is blank

**Comment by grzesuav** [comment]
> Thanks @grzesuav, I removed the whiteline. Do you see any additional work that needs to be done?

No, just the blank line

File: apis/metacontroller/v1alpha1/types.go
Comment (fpetkovski) [reply apis/metacontroller/v1alpha1/types.go]: Ah, the change wasn't staged. The blank line should be gone now.

**Review by grzesuav — APPROVED** [review]

**Pull request merged**
